DataProcessing/datax-host/src/main/scala/datax/fs/HadoopClient.scala

```diff
@@ -122,13 +122,21 @@ object HadoopClient {
     * @param sa name of the storage account
     */
   private def resolveStorageAccount(vaultName: String, sa: String) = {
+    // Fetch secret from keyvault using KeyVaultMsiAuthenticatorClient and if that does not return secret then fetch it using secret scope  
     val secretId = s"keyvault://$vaultName/${ProductConstant.ProductRoot}-sa-$sa"
```

Comment:
Curious to know. What if there is a key like this for a databricks environment? Will it stiill work?

Reply:
secret with name "${ProductConstant.ProductRoot}-sa-$sa" is created in keyvault. CP will try to first fetch the secret using KeyVaultMsiAuthenticatorClient which is used in HDInsight environment. For databricks environment, we read the same secret from the keyvault but the way to read a secret from keyvault in databrick is through secret scope. By adding the prefix 'secretscope' instead of 'keyvault', we tell CP to use secret scope through databricks utility to access the secret from keyvault since in databricks we cannot directly access secrets in keyvault